rust/publish-zenodo/src/lib.rs

```diff
@@ -0,0 +1,575 @@
+/// immediately, then use the `--force` flag.
+/// 
+/// By default, Zenodo's testing server is used. This means that DOIs created
+/// will not resolve. To upload to Zenodo's public-facing production server, use
+/// the `--zenodo` flag. To upload to a specific server instance, use
+/// `--server`.
+/// 
+/// Unless told otherwise, your deposit will be given the publication type "preprint".
+#[derive(Debug, Parser)]
+pub struct Cli {
+    /// Path to location of what to publish
+    #[arg(default_value = ".")]
+    #[arg(display_order(0))]
+    path: PathBuf,
+
+    /// Zenodo authentication token
+    /// 
+    /// To create one, log into Zenodo, visit your account's page, then click
+    /// "Applications", followed by "+ New Token" within the "Personal access
+    /// tokens" section. Give the token a name and enable the "deposit:actions" the scope.
+    /// 
+    /// Enable the "deposit:write" scope to enable the `--force` flag.
+    #[arg(long, env = "STENCILA_ZENODO_TOKEN")]
+    #[arg(help_heading("Zenodo Settings"))]
+    #[arg(display_order(1))]
+    token: Option<String>,
+
+    // Server selection options
+
+    /// Publish to Zenodo's testing server
+    #[arg(group = "zenodo_server")]
+    #[arg(long, default_value_t = true)]
+    #[arg(help_heading("Zenodo Settings"), display_order(1))]
+    sandbox: bool,
+
+    /// Publish to Zenodo's public-facing server
+    /// 
+    /// [default]
+    #[arg(group = "zenodo_server")]
+    #[arg(long)]
+    #[arg(help_heading("Zenodo Settings"), display_order(1))]
+    zenodo: bool,
+
+    /// Publish to a specific Zenodo server
+    #[arg(group = "zenodo_server")]
+    #[arg(long, value_parser = parse_host)]
+    #[arg(help_heading("Zenodo Settings"), display_order(1))]
+    server: Option<url::Host>,
+
+    // Resource type options
+
+    /// Upload document as a "lesson"
+    #[arg(group = "resource_type")]
+    #[arg(long)]
+    #[arg(conflicts_with_all = ["publication", "publication_type"])]
+    #[arg(help_heading("Deposition Settings"), display_order(2))]
+    lesson: bool,
+
+    /// Upload document as a "publication"
+    /// 
+    /// [default]
+    #[arg(group = "resource_type")]
+    #[arg(long, default_value_t = true)]
+    #[arg(default_value_if("lesson", common::clap::builder::ArgPredicate::IsPresent, "false"))]
+    #[arg(help_heading("Deposition Settings"), display_order(2))]
+    publication: bool,
+
+    /// Reserve a DOI for the deposition
+    #[arg(long)]
+    #[arg(help_heading("Deposition Settings"), display_order(2))]
+    reserve_doi: bool,
+
+    /// Publication date
+    /// 
+    /// When omitted, Zenodo will use today's date.
+    #[arg(long)]
+    #[arg(help_heading("Deposition Metadata"), display_order(3))]
+    #[arg(value_parser = parse_date)]
+    publication_date: Option<schema::Date>,
+
+    /// Title to use for the deposit
+    /// 
+    /// Required when the information is not available within the document.
```

Comment:
Maybe a message as to how you specify it in the document:

> Required when the information is not available within the document as an `Article:title` schema object.

Reply:
This will be useful, but it's not really specific enough to be actionable (yet).

We could add an example for YAML front matter in here. But that could potentially pollute the purpose of help string, which is to document the option's behaviour.

It would be really nice to point to some documentation. There is no documentation on the website or similar about what the schema is or how to add items to it. The instructions need to be specific for each input format.

Asking to wade through the `/docs` within the repo feels like a big task. And as it's just reference material, it's very difficult to interpret for beginners who are thinking "I just want to add a title to a document"

Reply:
Yes we absolutely need to get more guides and tutorials published on this soon.

Reply:
This should now be addressed by the long help text which is provided to people when they invoke `stencila help publish zenodo` or `stencila publish zenodo --help`.